pkgs/misc/cups/libcupsfilters.nix

```diff
@@ -0,0 +1,108 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, cups
+, dejavu_fonts
+, dbus
+, freetype
+, fontconfig
+, ghostscript
+, lcms2
+, libexif
+, libjpeg
+, libpng
+, libtiff
+, pkg-config
+, poppler_utils
+, mupdf
+, poppler
+, qpdf
+, withAvahi ? true
+, avahi
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libcupsfilters";
+  version = "2.0b2";
+
+  src = fetchFromGitHub {
+    owner = "OpenPrinting";
+    repo = "libcupsfilters";
+    rev = version;
+    hash = "sha256-jQtBuq3psdSw2UzMtj+wnxguAFPuXWNgSy1Awq5rkFs=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [
+    cups
+    libexif
+    libjpeg
+    libpng
+    libtiff
+    lcms2
+    freetype
+    fontconfig
+    qpdf
+    poppler
+    dbus
+    ghostscript
+    mupdf # used for mutool
+  ] ++ lib.optionals withAvahi [ avahi ];
+
+  configureFlags = [
+    "--with-mutool-path=${mupdf}/bin/mutool"
+    "--with-gs-path=${ghostscript}/bin/gs"
+    "--with-ippfind-path=${cups}/bin/ippfind"
```

Comment:
Maybe `lib.getBin` on these? libppd too.

Reply:
I'm not sufficiently proficient in Nix yet to comment on this. I've checked that `mupdf` and `lib.getBin mupdf` result in the same output paths (and similarly for `ghostscript` and `cups`). If it's considered best practice to use `lib.getBin` here I'm happy to change this, but I couldn't really find anything.

Reply:
I think `getBin` would be better in case one of the dependencies changes in the future, but it's a very marginal change for now.

Reply:
addressed in rebased commits